### PR TITLE
✨ Feat/profile

### PR DESCRIPTION
**Pull request by nataliadiaz2**
# Pull Request

## Story

Feature Edit Profile

## Description

Edit Profile form update, add profile elements, add styling to form 

## Analysis and Design

[Figma](https://www.figma.com/file/71cvT12tjZmTiwg6nRyuJb/AMA-Anytime?node-id=538%3A9757&t=XyaG0dPcvSwdUBJ4-0)
## Solution

Schema User update, GraphQL Profile update, Edit Profile update, 

## Affected Areas

User Table, Profile GraphQL, Edit Profile component

## Tests

Rendering correctly 

## Was this feature tested on all browsers?

- [X] Chrome
- [ ] Firefox
- [ ] Edge
- [ ] Internet Explorer
- [ ] Safari

## Definition of Done

- [ ] README updated
- [ ] Tests written
- [ ] Acceptance criteria implement
- [ ] Code reviewed
- [ ] Feature accepted

## PR Comments Emoji Legend

😻 - `:heart_cat_eyes:` Compliment to code/idea/etc
♻️ - `:recycle:` Non-blocking proposed refactor
➕ - `:heavy_plus_sign:` Non-blocking proposed addition
🔥 - `:fire:` Non-blocking proposed removal
❓ - `:question:` Non-blocking question
⚠️ - `:warning:` Blocking comment that must be addressed before PR

## Gifs for life (required)

mandatory

### CODE REVIEW

File: web/src/components/Avatar/Avatar.test.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    render(<Avatar {...AvatarNoImage} />)
+    expect(screen.getByTestId('initials')).toHaveTextContent('A')
+  })
+
+  it('displays the correct background color for the avatar', () => {
+    render(<Avatar {...AvatarNoImage} avatarColor="PUNCH" />)
+    expect(screen.getByTestId('avatar')).toHaveClass('PUNCH')
+  })
+  it('displays a default background color for the avatar', () => {
+    render(<Avatar {...AvatarNoImage} />)
```
Comment (archae0pteryx) [inline web/src/components/Avatar/Avatar.test.tsx:33]: I think these can all just be 2 tests. At least the tests that pass the same props (avatarNoImage) can all have their assertions in a single, "It renders default configuration / avatar / whatever". You can assert / expect more than one thing in a test if you wanna. Just a thought.

File: web/src/components/Avatar/Avatar.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  | 'PUNCH'
+  | 'ULCAGOLD'
+  | 'BLANC'
+  | 'ONYX'
+  | 'PESTO'
+  | 'FIORD'
+  | 'GREENSHEEN'
+  | 'VERIDIANGREEN'
+  | 'HEATWAVE'
+  | 'GRAY'
```
Comment (archae0pteryx) [inline web/src/components/Avatar/Avatar.tsx:11]: exactly what an enum is for.

**Review by archae0pteryx — COMMENTED** [review]
LGTM.

File: web/src/components/Avatar/Avatar.tsx [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+          className={`rounded-full ${className}`}
+          src={src}
+          alt={alt}
+          height={height}
+          width={width}
+        />
+      ) : (
+        <div
+          className={`flex items-center justify-center rounded-full text-center font-slab text-white ${className} ${
+            avatarColor || 'PUNCH'
```
Comment (craineum) [inline web/src/components/Avatar/Avatar.tsx:43]: ♻️ Instead defaulting to one of the items, this should maybe be a separate utility function that gets passed a string (name) and selects a color based on that name. That way you could also use the same thing for the cover background color.

**Review by craineum — CHANGES_REQUESTED** [review]
Looking good, some things to clean-up.

File: web/src/components/Profile/EditProfileCell/EditProfileCell.tsx [hunk 4]
```diff
@@ -67,21 +73,11 @@ export const Success = ({ profile }: CellSuccessProps) => {
     <>
       <MetaTags
         title={`${
-          profile.nickname || profile.name || profile.email
-        } | Edit Profile`}
+          profile.username || profile.name || profile.email
```
Comment (craineum) [inline web/src/components/Profile/EditProfileCell/EditProfileCell.tsx:76]: ♻️ Maybe this name selection process should be a utility function as well.

File: web/src/components/Profile/EditProfile/EditProfile.tsx [hunk 5]
```diff
@@ -1,80 +1,102 @@
...
+          <img
+            className="relative z-cover h-[295px] w-[96rem] object-cover object-center"
+            src={profile.cover}
+            alt={'cover'}
+            data-testid="cover"
+          />
+        ) : (
+          <img
+            className="relative h-[295px] w-[80rem] bg-gray object-cover object-center"
+            src="https://i.picsum.photos/id/237/750/295.jpg?hmac=G-dedeDLqcYUvDe-WUKTnnGIhDo6Kz5rzRwp-y6wgd4"
```
[the remark this answers is not in the document]
Comment (craineum) [reply web/src/components/Profile/EditProfile/EditProfile.tsx:27]: 💬 Let's pair on these utility functions today.

**Review by craineum — CHANGES_REQUESTED** [review]

File: web/src/utils/UserHelpers.ts [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+  'fiord',
+  'veridianGreen',
+  'heatWave',
+  'gray',
+]
+
+export const nameColor = (name) => {
+  const letterAPosition = 97
+  const letterNumber = name.toLowerCase().charCodeAt(0) - letterAPosition
+  return colors[letterNumber % 7]
```
Comment (craineum) [inline web/src/utils/UserHelpers.ts:16]: ♻️ Change the `7` to the length of colors.

**Review by craineum — APPROVED** [review]
One final tweak and it looks good to go!

**Pull request merged**